### PR TITLE
dont register invalid listen function

### PR DESCRIPTION
**Pull request by ghost**
if someone passes an invalid listener function, it will throw an error during key change,

`
const i = icaro()
i.listen({})
i.foo = bar
//TypeError: fn is not a function
`

### CODE REVIEW

File: icaro.js [hunk 1]
```diff
@@ -110,6 +110,9 @@ const API = {
    * @returns {API}
    */
   listen(fn) {
+    if(typeof fn !== "function"){
```
Comment (GianlucaGuarini) [inline icaro.js]: please use `if (typeof fn !== fn ) return this` to keep the api consistent and avoid the curly braces ;)

**Comment by GianlucaGuarini** [comment]
thank you it makes totally sense. Please fix the issues and the eslint errors and I will merge it

**Comment by ghost** [comment]
fixed linting errors.

File: icaro.js [hunk 2]
```diff
@@ -110,6 +110,7 @@ const API = {
    * @returns {API}
    */
   listen(fn) {
+    if(typeof fn !== 'function') return
```
Comment (GianlucaGuarini) [inline icaro.js]: here you need to `return this` to make the API consistent

Comment (ghost) [reply icaro.js]: ah sorry. i'll change it.

**Comment by GianlucaGuarini** [comment]
@anuragasaurus one last thing... could you add a test for this issue please?

**Comment by ghost** [comment]
yeah, sure

**Comment by GianlucaGuarini** [comment]
awesome thank you

**Pull request merged**